apps/choleskey/matrixutil.hpp

```diff
@@ -1,41 +1,163 @@
 #pragma once
 
-#include <iostream>
-#include <vector>
+#include <cblas.h>
+#include <lapacke.h>
+#include <stddef.h>
+#include <stdlib.h>
+#include <cstdlib>
+#include "commons.hpp"
+
+using data_type = double;
 
 // generate positive definition matrix
 template <typename T>
 using Matrix = std::vector<std::vector<T>>;
 
 template <typename T>
 std::vector<T> generate_pascal_matrix(const int n) {
-  Matrix<T> matrix(n, std::vector<T>(n, static_cast<T>(0)));
-
-  for (int i = 0; i < n; ++i) {
-    for (int j = 0; j < n; ++j) {
-      if (i == 0 || j == 0) {
-        matrix[i][j] = static_cast<T>(1);
-      } else {
-        matrix[i][j] = matrix[i][j - 1] + matrix[i - 1][j];
-      }
-    }
-  }
-
-  std::vector<T> flattenedVector;
-  for (const auto& row : matrix) {
-    flattenedVector.insert(flattenedVector.end(), row.begin(), row.end());
-  }
-  return std::move(flattenedVector);
+    Matrix<T> matrix(n, std::vector<T>(n, static_cast<T>(0)));
+
+    for (int i = 0; i < n; ++i) {
+        for (int j = 0; j < n; ++j) {
+            if (i == 0 || j == 0) {
+                matrix[i][j] = static_cast<T>(1);
+            } else {
+                matrix[i][j] = matrix[i][j - 1] + matrix[i - 1][j];
+            }
+        }
+    }
+
+    std::vector<T> flattenedVector;
```

Comment:
please change this `std::vector<T> flattenedVector;` to `std::vector<T> flattenedVector(N);` where N is your estimated biggest size of the flattened array size, so you can pre-allocate needed size.

Reply:
in this way, then you may not use `insert` in the following, maybe need to use `flattenedVector[i] = ...`

Reply:
Okay. As its not related in this PR, but useful for future clean up of previous code.  I write it down. thanks.